07-sampling.Rmd

```diff
@@ -211,7 +211,18 @@ Let's start by performing the virtual analog of the tactile sampling exercise we
 bowl
 ```
 
-Observe that `bowl` has 2400 rows, telling us that the bowl contains 2400 equally sized balls. The first variable `ball_ID` is used as an *identification variable* as discussed in Subsection \@ref(identification-vs-measurement-variables); none of the balls in the actual bowl are marked with numbers. The second variable `color` indicates whether a particular virtual ball is red or white. View the contents of the bowl in RStudio's data viewer and scroll through the contents to convince yourself that `bowl` is indeed a virtual analog of the actual bowl in Figure \@ref(fig:sampling-exercise-1).
+```{r message=FALSE, echo=FALSE, purl=FALSE}
+# Dynamic coding of summary statistics for bowl i.e. avoid hard-coding any values
+# wherever possible
+num_balls <- nrow(bowl)
+num_red <- bowl %>% 
+  summarize(red = sum(color == "red")) %>%
+  pull(red)
+prop_red <- num_red/num_balls
+percent_red_chr <- (100*prop_red) %>% str_c("%")
+```
+
+Observe that `bowl` has `r nrow(bowl)` rows, telling us that the bowl contains `r nrow(bowl)` equally sized balls. The first variable `ball_ID` is used as an *identification variable* as discussed in Subsection \@ref(identification-vs-measurement-variables); none of the balls in the actual bowl are marked with numbers. The second variable `color` indicates whether a particular virtual ball is red or white. View the contents of the bowl in RStudio's data viewer and scroll through the contents to convince yourself that `bowl` is indeed a virtual analog of the actual bowl in Figure \@ref(fig:sampling-exercise-1).
```

Comment:
I added this code chunk here, b/c it is right after `bowl` is introduced to reader.